herbert_core/applications/multifit/@mfclass/fit.m

```diff
@@ -111,57 +103,45 @@
         elseif sum(lopt)==1
             output_type = flagnames{lopt};
         else
-            ok = false; mess = 'Check the value of output option';
+            error("Herbert:mfclass:badargs", 'Check the value of output option')
         end
     else
-        ok = false; mess = 'Check number of input arguments';
+        error("Herbert:mfclass:badargs", 'Check number of input arguments')
     end
-end
-if ~ok
-    if throw_error, error_message(mess), else, return, end
+else
+    error("Herbert:mfclass:badargs", mess)
 end
 
 % Check that there is data present
 if obj.ndatatot_ == 0
-    ok = false; mess = 'No data has been provided for fitting';
-    if throw_error, error_message(mess), else, return, end
+    error("Herbert:mfclass:badargs", 'No data has been provided for fitting')
 end
 
 % Check that all functions are present
 foreground_present = ~all(cellfun(@isempty,obj.fun_));
 background_present = ~all(cellfun(@isempty,obj.bfun_));
+
 if ~foreground_present && ~background_present
-    ok = false; mess = 'No fit functions have been provided';
-    if throw_error, error_message(mess), else, return, end
+    error("Herbert:mfclass:badargs", 'No fit functions have been provided')
```

Comment:
[ADR 21](https://github.com/pace-neutrons/Horace/blob/master/documentation/adr/0021-errors-and-warnings.md#recommendations) suggests the first two parts of the error ID should be upper case